### PR TITLE
Bundle test framework adapters into PHAR

### PR DESCRIPTION
**Pull request by maks-rafalko**
This PR:

- [x]  closes https://github.com/infection/infection/issues/1129
- [x] bundles adapters into PHAR
- [x] Doc PR: https://github.com/infection/site/pull/159

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -185,8 +185,10 @@ $(PHP_CS_FIXER): Makefile
 $(PHPSTAN): vendor
 
 $(INFECTION): vendor $(shell find bin/ src/ -type f) $(BOX) box.json.dist .git/HEAD
+	composer require infection/codeception-adapter infection/phpspec-adapter
 	$(BOX) validate
 	$(BOX) compile
+	composer remove infection/codeception-adapter infection/phpspec-adapter
```
Comment (maks-rafalko) [inline Makefile]: without removing adapters, Travis fails with weird errors during `bin/infection ...`

I wasn't able understand why

* locally works as expected
* tried debugging by ssh'ing into container im Travis - random fails

Comment (sanmai) [reply Makefile]: It says InitialTestsFailed. Can we assume that this is a mistake? E.g. tests are not broken.

Comment (maks-rafalko) [reply Makefile]: ok, I admit that the machine is smarter than me. It fails locally as well with unit tests for known reasons. Tests assume that classes from adapters do not exist.

Anyway, removing adapters is ok for me here.

**Comment by maks-rafalko** [comment]
Thank you @sanmai & @theofidry

**Pull request merged**